### PR TITLE
feat(explore): Implement data table empty states

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
This PR implements chart empty states for 3 scenarios:

1) Not all required controls have values
2) Query returned no results
3) No samples were retrieved for dataset

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
image

image

image

### TESTING INSTRUCTIONS
1. Create a chart
2. Run a query that returns 0 results (for example set time range for last 5 minutes)
3. Remove required controls

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

CC @kasiazjc

https://app.shortcut.com/preset/story/36403/explore-empty-states

### CODE REVIEW

File: superset-frontend/src/explore/components/DataTablesPane/index.tsx [hunk 1]
```diff
@@ -152,14 +155,19 @@ const DataTable = ({
   }
   if (data) {
     if (data.length === 0) {
-      return <span>No data</span>;
+      return (
+        <EmptyStateMedium
+          image="document.svg"
+          title={t(`No ${type} were returned for this query`)}
```
Comment (villebro) [inline superset-frontend/src/explore/components/DataTablesPane/index.tsx:161]: I think we need to translate this like this:
```suggestion
          title={t(`No %s were returned for this query`, type)}
```

File: superset-frontend/src/explore/components/DataTablesPane/index.tsx [hunk 2]
```diff
@@ -169,7 +177,12 @@ const DataTable = ({
     );
   }
   if (errorMessage) {
-    return <Error>{errorMessage}</Error>;
+    return (
+      <EmptyStateMedium
+        image="document.svg"
+        title={t(`Run a query to display ${type}`)}
```
Comment (villebro) [inline superset-frontend/src/explore/components/DataTablesPane/index.tsx:183]: Same here

**Review by villebro — COMMENTED** [review]
Two translation comments, other than that LGTM

File: superset-frontend/src/explore/components/DataTablesPane/index.tsx (hunk 1 above)
Comment (villebro) [reply superset-frontend/src/explore/components/DataTablesPane/index.tsx:161]: Actually, we probably need to translate these like this to make sure the translators have the necessary context to fully translate the type strings:

```typescript
if (type === 'sample') {
  return t(`No samples were returned for this query`)
}
if (type === 'result') {
  return t(`No results were returned for this query`)
}
```
I think this getting this right is pretty important, as these will be very visible fixtures in the application.

Comment (kgabryje) [reply superset-frontend/src/explore/components/DataTablesPane/index.tsx]: Done

File: superset-frontend/src/explore/components/DataTablesPane/index.tsx (hunk 2 above)
Comment (kgabryje) [reply superset-frontend/src/explore/components/DataTablesPane/index.tsx]: Done

**Review by villebro — APPROVED** [review]

**Pull request merged**